### DISCUSSION TITLE
Trouble making a spectrum

### DISCUSSION THREAD

**Discussion by jonhulberg** [dullemond/radmc3d-2.0#35] [Q&A]
Hello, I'm a PHD student working on modeling dusty CSEs around AGB stars. I've made a simple 1D model with a spherically symmetric gas and dust density,  and am trying to create a spectrum with line radiative transfer turned on. I used the command `radmc3d spectrum lambdarange 1. 25. nlam 100`. This returned the error message `INTERNAL ERROR in lines: lines_maxshift variable not set`.  I figured I was doing something wrong, so I checked the simple model in run_lines_lte_0d_1. I found that calling `radmc3d spectrum lambdarange ....` also returns the same error message for this example. As far as I can tell, lines_maxshift is a variable in the lines.f90 file. Is anyone familiar with this error?

Thanks,
Jon Hulberg

**Comment by dullemond** [comment]
Dear Jon,
admittedly this is a bit a quirk of RADMC-3D. If you want to do radiative transfer of gas lines, you have to select the line you wish to model, as in `radmc3d spectrum iline 1 widthkms 1`. If you use `radmc3d spectrum lambdarange 1. 25. nlam 100` that would be like an SED, i.e. a very broad spectrum that only makes sense for dust continuum radiative transfer. Reason: the width of a gas line is super-narrow compared to the wavelength range 1-25 micron. When you simply do RT at 100 wavelength points between 1 and 25 micron, then you will never 'hit' the right wavelength of a gas line. So the gas lines won't show up in the spectrum. If, by incredible chance, you _do_ hit a line, then it will artificially have the width of a wavelength sampling interval (roughly 0.1 micron), whereas the line is in reality much narrower. The line will then artificially get a way too large flux. So either way it is bad: Either you do not see any lines, or, if you do, they are way too luminous. 

So, whenever you do `radmc3d spectrum lambdarange 1. 25. nlam 100`, you want to only treat the dust continuum. To do this, better switch off the line stuff with the command `nolines`. In other words: `radmc3d spectrum lambdarange 1. 25. nlam 100 nolines`. 

If you want to model a line, then better use `radmc3d spectrum iline 1 widthkms 1` (or something like that).

Hope this helps.

Best wishes,
Kees

**Comment by jonhulberg** [comment]
Hi Kees,

Thanks for explaining, this makes sense now.

Jon
